internal/fleet/agent_policy_resource.go

```diff
@@ -189,6 +241,25 @@ func resourceAgentPolicyUpdate(ctx context.Context, d *schema.ResourceData, meta
 	}
 	req.MonitoringEnabled = &monitoringValues
 
+	if tags, ok := d.GetOk("global_data_tags"); ok {
+		tagMap := tags.(map[string]interface{})
+
+		if len(tagMap) > 0 && serverVersion.LessThan(minVersionGlobalDataTags) {
+			return diag.FromErr(fmt.Errorf("'global_data_tags' is supported only for Elasticsearch v%s and above", minVersionGlobalDataTags.String()))
+		}
+
+		globalDataTags := make([]fleetapi.GlobalDataTag, 0, len(tagMap))
+		for key, value := range tagMap {
+			globalDataTags = append(globalDataTags, fleetapi.GlobalDataTag{
+				Name:  key,
+				Value: value.(string),
+			})
+		}
+		req.GlobalDataTags = globalDataTags
+	} else {
+		req.GlobalDataTags = make([]fleetapi.GlobalDataTag, 0) // Ensure it's an empty array
```

Comment:
Similarly, why do we need this to be explicitly empty?

Reply:
see above